internal/controllers/user_integrations_controller.go

```diff
@@ -58,14 +58,94 @@ func (udc *UserDevicesController) GetUserDeviceIntegration(c *fiber.Ctx) error {
 	apiIntegration, err := models.UserDeviceAPIIntegrations(
 		models.UserDeviceAPIIntegrationWhere.UserDeviceID.EQ(userDeviceID),
 		models.UserDeviceAPIIntegrationWhere.IntegrationID.EQ(integrationID),
+		qm.Load(models.UserDeviceAPIIntegrationRels.UserDevice),
 	).One(c.Context(), udc.DBS().Reader)
 	if err != nil {
 		if errors.Is(err, sql.ErrNoRows) {
 			return fiber.NewError(fiber.StatusBadRequest, fmt.Sprintf("user device %s does not have integration %s", userDeviceID, integrationID))
 		}
 		return err
 	}
-	return c.JSON(GetUserDeviceIntegrationResponse{Status: apiIntegration.Status, ExternalID: apiIntegration.ExternalID, CreatedAt: apiIntegration.CreatedAt})
+
+	resp := GetUserDeviceIntegrationResponse{
+		Status:     apiIntegration.Status,
+		ExternalID: apiIntegration.ExternalID,
+		CreatedAt:  apiIntegration.CreatedAt,
+	}
+	// Handle fetching virtual token status
+	intd, err := udc.DeviceDefSvc.GetIntegrationByID(c.Context(), integrationID)
+	if err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, "invalid integration id")
+	}
+	if intd.Vendor == constants.TeslaVendor {
+		if !apiIntegration.ExternalID.Valid && apiIntegration.AccessToken.Valid && !(apiIntegration.R.UserDevice.VinConfirmed && apiIntegration.R.UserDevice.VinIdentifier.Valid) {
```

Comment:
What if the Tesla is on the old API (V1)?

Reply:
This conditional looks a bit odd to me. We're erroring if

```
no external id AND there's an access token
                        AND (the VIN isn't marked unconfirmed OR there's no VIN)
```

It seems like any one of these things (no external id, no access token, no VIN, VIN unconfirmed) is cause for alarm.